### PR TITLE
Core/Refactor: Split Position in it's own file.

### PR DESCRIPTION
**Pull request by FALL1N1**
## CHANGES PROPOSED:
- Move the Position struct outside Object so it's easier to use, this PR is a "dependancy" for some new ones that's coming out.

## ISSUES ADDRESSED:
- none

## TESTS PERFORMED:
- none, not necessary 

## KNOWN ISSUES AND TODO LIST:

- none

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FALL1N1** [comment]
WIP tag pls <3

File: src/server/game/Entities/Object/Object.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include "World.h"
 #include "Object.h"
 #include "Creature.h"
-#include "Player.h"
+#include "Player.h"
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Object/Object.cpp:15]: revert this pls

**Comment by Si1ker** [comment]
@FALL1N1 👀  ?

**Comment by FALL1N1** [comment]
@Si1ker this PR is no longer necessary, I'll close it.

**Pull request closed without merging**